lib/internal/modules/esm/resolve.js

```diff
@@ -463,6 +463,21 @@ const patternRegEx = /\*/g;
 
 function resolvePackageTargetString(
   target, subpath, match, packageJSONUrl, base, pattern, internal, conditions) {
+
+  const composeResult = (resolved) => {
+    let format;
+    try {
+      format = getPackageType(resolved);
+    } catch (err) {
+      if (err.code === 'ERR_INVALID_FILE_URL_PATH') {
+        throw new ERR_INVALID_MODULE_SPECIFIER(
+          resolved, 'must not include encoded "/" or "\\" characters', base);
+      }
+      throw err;
```

Comment:
why create a new error?

Reply:
that is to satisfy the assertion in the existing testcases (and avoid any undesired side-effects): https://github.com/nodejs/node/blob/8ac9aef8b1a323399ed1d67fb3a90c6c4d9e4d7a/test/es-module/test-esm-imports.mjs#L69